tests/gtest/avifheadertest.cc

```diff
@@ -0,0 +1,39 @@
+// Copyright 2024 Google LLC
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <algorithm>
+
+#include "avif/avif.h"
+#include "aviftest_helpers.h"
+#include "gtest/gtest.h"
+
+namespace avif {
+namespace {
+
+TEST(BasicTest, EncodeDecode) {
+  ImagePtr image = testutil::CreateImage(12, 34, 8, AVIF_PIXEL_FORMAT_YUV420,
+                                         AVIF_PLANES_ALL);
+  ASSERT_NE(image, nullptr);
+  testutil::FillImageGradient(image.get());
+  const testutil::AvifRwData avif = testutil::Encode(image.get());
+  ASSERT_NE(avif.data, nullptr);
+
+  // Make sure the HandlerBox is as small as possible, meaning its name field is
+  // empty.
+  const uint8_t* kHdlr = reinterpret_cast<const uint8_t*>("hdlr");
```

Comment:
Nit: The way `kHdlr` is defined here, it is a variable, not a constant. (It is not even a const variable.) So it should be named like a variable:

```
    const uint8_t* const hdlr = reinterpret_cast<const uint8_t*>("hdlr");
```

Alternatively, we can define it as a constant:

```
    const uint8_t kHdlr[] = { 'h', 'd', 'l', 'r' };
```